samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,921 @@
+
+	pipeline_create_info.pVertexInputState = &vertex_input_state_create_info;
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipeline_gui));
+}
+
+// Prepare and initialize uniform buffer containing shader uniforms
+void DynamicMultisampleRasterization::prepare_uniform_buffers()
+{
+	// Matrices vertex shader uniform buffer
+	uniform_buffer = std::make_unique<vkb::core::BufferC>(get_device(),
+	                                                      sizeof(uniform_data),
+	                                                      VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+	                                                      VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	update_uniform_buffers();
+}
+
+void DynamicMultisampleRasterization::prepare_gui()
+{
+	create_gui(*window, /*stats=*/nullptr, 15.0f, true);
+
+	prepare_gui_pipeline();
+
+	// No need to call gui->prepare because the pipeline has been created above
+}
+
+void DynamicMultisampleRasterization::update_uniform_buffers()
+{
+	uniform_data.projection = camera.matrices.perspective;
+	// Scale the view matrix as the model is pretty large, and also flip it upside down
+	uniform_data.view = glm::scale(camera.matrices.view, glm::vec3(0.1f, -0.1f, 0.1f));
+	uniform_buffer->convert_and_update(uniform_data);
+}
+
+void DynamicMultisampleRasterization::draw()
+{
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+void DynamicMultisampleRasterization::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+	draw();
+
+	if (camera.updated)
+	{
+		update_uniform_buffers();
+	}
+}
+
+void DynamicMultisampleRasterization::destroy_image_data(ImageData &image_data)
+{
+	if (image_data.image != nullptr)
+	{
+		vkDestroyImageView(get_device().get_handle(), image_data.view, nullptr);
+		vkDestroyImage(get_device().get_handle(), image_data.image, nullptr);
+		vkFreeMemory(get_device().get_handle(), image_data.mem, nullptr);
+		image_data.view  = VK_NULL_HANDLE;
+		image_data.image = VK_NULL_HANDLE;
+		image_data.mem   = VK_NULL_HANDLE;
+	}
+}
+
+void DynamicMultisampleRasterization::update_resources()
+{
+	prepared = false;
+
+	if (has_device())
+	{
+		get_device().wait_idle();
+		rebuild_command_buffers();
+	}
+
+	prepared = true;
+}
+
+void DynamicMultisampleRasterization::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	if (drawer.header("Settings"))
+	{
+		if (drawer.combo_box("antialiasing", &gui_settings.sample_count_index, gui_settings.sample_counts))
+		{
+			sample_count = supported_sample_count_list[gui_settings.sample_count_index];
+
+			update_resources();
+		}
+	}
+}
+
+bool DynamicMultisampleRasterization::resize(const uint32_t _width, const uint32_t _height)
+{
+	sample_count = VK_SAMPLE_COUNT_1_BIT;
```

Comment:
Some comment why you need to switch sample_count to 1_BIT here and back to the "actual" value right after resizing would be fine.

Reply:
This is related to this VL Error

`Validation Error: [ VUID-VkFramebufferCreateInfo-pAttachments-00881 ] Object 0: handle = 0xba7514000000002a, type = VK_OBJECT_TYPE_RENDER_PASS; Object 1: handle = 0x2002ea00000002ad, type = VK_OBJECT_TYPE_IMAGE_VIEW; Object 2: handle = 0x46b46400000002ab, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0x2ff52eec | vkCreateFramebuffer(): pCreateInfo->pAttachments[1] has VK_SAMPLE_COUNT_4_BIT samples that do not match the VK_SAMPLE_COUNT_1_BIT samples used by the corresponding attachment for VkRenderPass 0xba7514000000002a[].`

While window is resized recreation of framebuffer is performing. When using other sample count then `VK_SAMPLE_COUNT_1_BIT` it is complaining because renderpass sample count is by default set to `VK_SAMPLE_COUNT_1_BIT`

I can think about more neat solution, but if needed only :)